src/Workspaces/Core/Portable/Workspace/Solution/SolutionCompilationState.CompilationTracker.cs

```diff
@@ -124,53 +124,61 @@ SymbolKind.NetModule or
                     ForkState(oldProjectState: this.ProjectState, ReadState(), translate),
                     this.SkeletonReferenceCache.Clone());
 
-                static CompilationTrackerState? ForkState(
+                static InProgressState? ForkState(
                     ProjectState oldProjectState,
                     CompilationTrackerState? state,
                     CompilationAndGeneratorDriverTranslationAction? translate)
                 {
                     if (state is null)
                         return null;
 
+                    // Determine the old/stale compilation to help seed the non-final state with.
+                    var staleCompilationWithGeneratedDocuments = state switch
+                    {
+                        FinalState finalState => finalState.FinalCompilationWithGeneratedDocuments,
+                        InProgressState inProgressState => inProgressState.StaleCompilationWithGeneratedDocuments,
+                        _ => throw ExceptionUtilities.UnexpectedValue(state.GetType()),
+                    };
+
+                    var intermediateProjects = UpdateIntermediateProjects(oldProjectState, state, translate);
+
+                    var newState = InProgressState.Create(
+                        state.IsFrozen,
+                        state.CompilationWithoutGeneratedDocuments,
+                        state.GeneratorInfo,
+                        staleCompilationWithGeneratedDocuments,
+                        intermediateProjects);
+                    return newState;
+                }
+
+                static ImmutableList<(ProjectState oldState, CompilationAndGeneratorDriverTranslationAction action)> UpdateIntermediateProjects(
```

Comment:
>IntermediateProjects [](http://example.com/codeflow?start=123&length=20)

The IntermediateProjects name seems a bit off to me. It's a (projectstate, action) pair, which is really more like a translation that has some added state to it in my mind.

Reply:
i agree. i was just keepign the name consistent.  i'm happy to rename in a followup.  'StatesAndTranslationActions might be good.